spec/lib/context_request_middleware/middleware_spec.rb

```diff
@@ -99,6 +99,32 @@ module ContextRequestMiddleware
           subject.call(env)
         end
       end
+
+      context 'with missing HTTP_X_REQUEST_START header' do
```

Comment:
You are not testing the case when you fall back to ``Time.current || Time.now`` right? But I think this case should be tested.